api/transport/transporttest/reqres.go

```diff
@@ -73,6 +73,21 @@ func (m RequestMatcher) Matches(got interface{}) bool {
 		panic(fmt.Sprintf("expected *transport.Request, got %v", got))
 	}
 
+	if l.ID != r.ID {
```

Comment:
question: do we still need match check in transport or environment-aware middleware layer?

Reply:
Good questions:

> does that means we left the check logic to client

yes!

> NewRequestMatcher is only called in test functions

`RequestMatcher` lives within the `transporttest` package and is only intended for use within tests. It's purpose is to match two `transport.Request`s (not a request/response pair).

> do we still need match check in transport or environment-aware middleware layer?

Yes -- assuming you're referring to matching service name and ID transport headers like you did in your PRs. These additional fields enable us to consolidate validation logic from within multiple transports to one spot (potentially the `validator_outbound` or some middleware) that understands the common transport types: `transport.Request`/`transport.Response`

Reply:
more clear now, thanks for explanation